packages/wrangler/src/pages/upload.tsx

```diff
@@ -200,16 +200,21 @@ export const upload = async (
 		const doUpload = async (): Promise<void> => {
 			// Populate the payload only when actually uploading (this is limited to 3 concurrent uploads at 50 MiB per bucket meaning we'd only load in a max of ~150 MiB)
 			// This is so we don't run out of memory trying to upload the files.
-			const payload: UploadPayloadFile[] = await Promise.all(
-				bucket.files.map(async (file) => ({
-					key: file.hash,
-					value: (await readFile(file.path)).toString("base64"),
-					metadata: {
-						contentType: file.contentType,
-					},
-					base64: true,
-				}))
-			);
+			const payload: UploadPayloadFile[] = [];
+
+			for (let i = 0; i < bucket.files.length; i += 1000) {
+				// upload files 1k at once to avoid EMFILE
```

Comment:
I think what we mean here is:

```suggestion
				// only read up to 1000 files, from disk, at a time to avoid `EMFILE` error (on Windows)
```

Reply:
Should we just consider using graceful-fs instead?

Reply:
Maybe, but:

1) this avoids a dep
2) this is faster (presumably) and more obviously correct

The only case that handles better is if there's a weird OS where the limit on the # of file descriptors is <1k which is not something to be worried about IMO. But we can use `graceful-fs` here ultimately, I'm not sure how to weigh the pros vs cons.